staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_whitebox_test.go

```diff
@@ -2470,6 +2470,72 @@ func TestWatchStreamSeparation(t *testing.T) {
 	}
 }
 
+func TestDispatchEventsUseProgressRequester(t *testing.T) {
+	featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.ConsistentListFromCache, true)
+	featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.WatchList, true)
+
+	backingWatch := &dummyWatch{ch: make(chan watch.Event, 1)}
+	backingStorage := &dummyStorage{}
+	backingStorage.watchFn = func(_ context.Context, _ string, _ storage.ListOptions) (watch.Interface, error) {
+		return backingWatch, nil
+	}
+	cacher, _, err := newTestCacher(backingStorage)
+	require.NoError(t, err)
+	defer cacher.Stop()
+
+	// wait until cacher is initialized.
+	if err := cacher.ready.wait(context.Background()); err != nil {
+		t.Fatalf("unexpected error waiting for the cache to be ready")
+	}
+	require.Equal(t, 0, backingStorage.getRequestWatchProgressCounter(), "unexpected value of requestWatchProgressCounter")
+
+	// start a watch without sending
+	// the bookmark even from the backing storage
+	pred := storage.Everything
+	pred.AllowWatchBookmarks = true
+	opts := storage.ListOptions{
+		Predicate:         pred,
+		SendInitialEvents: pointer.Bool(true),
+	}
+	w, err := cacher.Watch(context.Background(), "pods/ns", opts)
+	require.NoError(t, err, "failed to create watch: %v")
+	testCheckNoEvents(t, w)
+	w.Stop()
+	storeWatchProgressCounterValueAfterFirstWatch := backingStorage.getRequestWatchProgressCounter()
```

Comment:
maybe before stopping the watch we should loop until the counter will be > 0 ? That could deflake the test